### PR TITLE
DOC: First steps towards docs restructuring (NEP 44)

### PR DESCRIPTION
**Pull request by melissawm**
Hi all,

This PR implements the first steps of the documentation restructure outlined in [NEP 44](https://numpy.org/neps/nep-0044-restructuring-numpy-docs.html). Some placeholders were added, but I plan to add real content to them soon. 

Thanks!

### CODE REVIEW

File: doc/source/contents.rst [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
 .. toctree::
 
-   user/index
+   user/setting-up
+   user/quickstart
+   user/absolute_beginners
+   user/tutorials_index
+   user/howtos_index
    reference/index
+   user/explanations_index
```
Comment (charris) [inline doc/source/contents.rst]: Any reason not to group the major divisions together?

**Comment by charris** [comment]
I think `development` would be a better name than `dev`?

There is a polynomial tutorial `doc/source/reference/routines.polynomials.classes.rst`. I'm not sure how to reorganize that as the classes themselves are included in the appropriate polynomial modules and documented with them even though they all have the same methods, so only need to be documented once. Hmm. Is there a way to keep certain objects from being documented by default? In fact, most of the functions in the different modules implement type specific versions of the same functionality and have names the differ only by a prefix. The polynomial documentation takes up altogether to much space, it would be good to figure out how to reduce the amount of repetition. That doesn't need to be done here, but it would be nice to think about how to do it.

**Comment by rossbar** [comment]
@charris - as you say, this discussion may belong elsewhere, but since you mention it here I wanted to give a quick accounting of my thinking having read my way through the `polynomial` package & docs. One approach that makes sense to me is to centralize the documentation at the package-level - i.e. include a lot of the relevant material from `.classes.rst` (which describe the preferred interface) to the polynomial package doc string. `np.polynomial` is the namespace that provides the preferred interface (i.e. the convenience classes) to all of the types of polynomial, while the submodules (e.g. `np.polynomial.polynomial`) are mostly implementation details. I think concentrating the documentation around and pointing to the `np.polynomial` package itself could really help resolve some of the issues surrounding the confusing naming conventions in the package.

This is likely a much longer discussion and I don't want to completely hi-jack the thread. Note also that #15662 is largely about trying to deal with the volume & redundancy of the `np.polynomial` documentation and represents a first step in the direction you propose.

**Comment by charris** [comment]
Lets put this in, we can nitpick the details later. Thanks @melissawm .

File: doc/source/user/ionumpy.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+**Objectives**
+
+- Writing NumPy arrays to files
+- Reading NumPy arrays from files
+- Dealing with encoding and dtype issues
+
+**Content**
+
+To be completed.
```
Comment (rgommers) [inline doc/source/user/ionumpy.rst]: @melissawm I think you just added one so you could put in a `toctree`? In that case this seems fine, and this document can be written soon. It'd be best not to put more placeholders in though imho, better to use issues for that.

**Comment by rgommers** [comment]
Thanks @melissawm. This looks good to me!

File: doc/source/user/ionumpy.rst (hunk 2 above)
Comment (melissawm) [reply doc/source/user/ionumpy.rst]: That's right, I just wanted it to be in the toctree. I agree about no more placeholders .

**Pull request merged**